CDCgov/NEDSS-Modernization
# CNFT2-1185 related units

## Description

A bug was created for some fields were missing from the create question API. It appears the fields are there just not apparent. 

Updated the Numeric question model to remove the enum and add a `Long relatedUnitsValueSet` and a `String relatedUnitsLiteral` to better signal the intended usage.


## Tickets

* [CNFT2-1185](https://cdc-nbs.atlassian.net/browse/CNFT2-1185)

## Steps to verify

1. Login and get a token via the modernization-api
1. Send a request to create a new Numeric question via [page builder swagger](http://localhost:8095/nbs/page-builder/swagger-ui/#/question-controller/createQuestionUsingPOST)
```json
{
  "type": "NUMERIC",
    "adminComments": "Numeric Question administrative comments",
    "codeSet": "LOCAL",
    "dataMartInfo": {
        "dataMartColumnName": "NUM_DATA_MART_COLUMNA",
        "defaultRdbTableName": "D_INV_ADMINISTRATIVEA",
        "rdbColumnName": "NUM_RDB_COLUMN_NAMEA",
        "reportLabel": "Numeric report label 2"
    },
    "description": "Test Numeric Question",
    "displayControl": 1008,
    "label": "Sample Numeric question label 2",
    "allowFutureDates": false,
    "messagingInfo": {
        "codeSystem": "ABNORMAL_FLAGS_HL7",
        "hl7DataType": "CE",
        "includedInMessage": true,
        "labelInMessage": "Numeric label in message",
        "messageVariableId": "Numeric MessageVariableId",
        "requiredInMessage": true
    },
    "subgroup": "ADM",
    "tooltip": "Sample numeric question tooltip",
    "uniqueId": null,
    "uniqueName": "Sample Numeric question 2",
    "mask": "NUM",
    "fieldLength": "3",
    "defaultValue": "1",
    "minValue": 0,
    "maxValue": 100,
    "relatedUnitsLiteral": "some literal value"
}
```
3. Verify the question is created and that its `unit_value` column is set to `some literal value` and `unit_type_cd` is set to `LITERAL`. The Classic™ UI should reflect this as well
![image](https://github.com/CDCgov/NEDSS-Modernization/assets/109251240/5cbf17ce-7cf7-493e-9340-b86e12df7c5f)

5. Delete the question from the DB
6. Send another a request to create a new Numeric question with a resulted units value set
```json
{
  "type": "NUMERIC",
    "adminComments": "Numeric Question administrative comments",
    "codeSet": "LOCAL",
    "dataMartInfo": {
        "dataMartColumnName": "NUM_DATA_MART_COLUMNA",
        "defaultRdbTableName": "D_INV_ADMINISTRATIVEA",
        "rdbColumnName": "NUM_RDB_COLUMN_NAMEA",
        "reportLabel": "Numeric report label 2"
    },
    "description": "Test Numeric Question",
    "displayControl": 1008,
    "label": "Sample Numeric question label 2",
    "allowFutureDates": false,
    "messagingInfo": {
        "codeSystem": "ABNORMAL_FLAGS_HL7",
        "hl7DataType": "CE",
        "includedInMessage": true,
        "labelInMessage": "Numeric label in message",
        "messageVariableId": "Numeric MessageVariableId",
        "requiredInMessage": true
    },
    "subgroup": "ADM",
    "tooltip": "Sample numeric question tooltip",
    "uniqueId": null,
    "uniqueName": "Sample Numeric question 2",
    "mask": "NUM",
    "fieldLength": "3",
    "defaultValue": "1",
    "minValue": 0,
    "maxValue": 100,
    "relatedUnitsValueSet": 1760
}
```
7. Verify the question is created and that its `unit_value` column is set to `1760` and `unit_type_cd` is set to `CODED` and that the Classic™ UI shows the value set
![image](https://github.com/CDCgov/NEDSS-Modernization/assets/109251240/356d3f31-0cc9-4e19-94ce-6c7fe96236e7)



[CNFT2-1185]: https://cdc-nbs.atlassian.net/browse/CNFT2-1185?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
